docs/specs/devcontainer-templates.md

```diff
@@ -66,6 +67,26 @@ The `options` property contains a map of option IDs and their related configurat
 
 > `Note`: The `options` must be unique for every `devcontainer-template.json`
 
+### The `optionalPaths` property
+
+Before applying a Template, tooling must inspect the `optionalPaths` property of a Template and prompt the user on whether each file or folder should be included in the resulting output workspace folder.  
+
+An optional path is relative to the root of the Template source directory.  For a directory, a trailing slash (`/`) should be appended to the path.  Examples are shown below:
+
+```json
+{
+    "id": "cpp",
+    "version": "3.0.0",
+    "name": "C++",
+    "description": "Develop C++ applications",
+    "optionalPaths": [
+         "GETTING-STARTED.md",
+         "example-project-1/MyProject.csproj",
+         ".github/"
```

Comment:
I wonder if folks might forget to add the `/` suffix to denote a folder. Should we allow `/*` instead, which seems more familiar (at least to me)? However, I am definitely open to your thoughts on this 👀

Reply:
Thanks for the feedback! 

> I wonder if folks might forget to add the / suffix to denote a folder and the CLI might incorrectly work?

In my tagged implementation omitting the `/` wouldn't cause any of its child files to be included, so it would work as intended

> Should we allow /* instead, which seems more familiar 

Sounds reasonable to me.  I don't think we have such a precedent in the dev containers spec for how this should look.  

Though, perhaps i'm overthinking it, as all of `.github`, `.github/`, and `.github/*` would get copied with the `cp -r` command. 

Another way this could be phrased is `optionalPathPrefixes`, where anything under the prefix gets copied (that's essentially what is happening here but always with an implicit `/*` at the end matching anything that shares that path prefix.